src/org/traccar/database/LdapProvider.java

```diff
@@ -16,36 +16,161 @@
 package org.traccar.database;
 
 import javax.naming.Context;
+import javax.naming.NamingEnumeration;
 import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
 import javax.naming.directory.InitialDirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+
+import org.traccar.Config;
+import org.traccar.helper.Log;
+import org.traccar.model.User;
+
 import java.util.Hashtable;
 
 public class LdapProvider {
 
     private String url;
-    private String context;
+    private String searchBase;
+    private String idAttribute;
+    private String nameAttribute;
+    private String mailAttribute;
+    private String searchFilter;
+    private String adminFilter;
+    private String serviceUser;
+    private String servicePassword;
 
-    public LdapProvider(String url, String context) {
-        this.url = url;
-        this.context = context;
+    public LdapProvider(Config config) {
+        String url = config.getString("ldap.url");
+        if (url != null) {
+            this.url = url;
+        } else {
+            this.url = "ldap://" + config.getString("ldap.server") + ":" + config.getInteger("ldap.port", 389);
+        }
+        this.searchBase = config.getString("ldap.base");
+        this.idAttribute = config.getString("ldap.idAttribute", "uid");
+        this.nameAttribute = config.getString("ldap.nameAttribute", "cn");
+        this.mailAttribute = config.getString("ldap.mailAttribute", "mail");
+        this.searchFilter = config.getString("ldap.searchFilter", "(" + idAttribute + "=:login)");
```

Comment:
It is customized search filter to pre lookup user, `:login` replaced by searching login, by default it is `(uid=:login)` or `(sAMAccountName=:login)`
Here is couple of examples:
- `(|(sAMAccountName=:login)(mail=:login))` login by account name or email address
- `(&(sAMAccountName=:login)(memberOf:1.2.840.113556.1.4.1941:=CN=Traccar users,OU=Traccar,OU=Security,DC=domain,DC=local)(!(UserAccountControl:1.2.840.113556.1.4.803:=2)))` user must be member of `Traccar users` group or its nested groups (!) and not disabled. Valid for Active Directory